src/Configuration.php

```diff
@@ -129,18 +154,45 @@ public function getDOMDocument() {
      * @param  string     $xpathExpression
      * @param  boolean    $alwaysReturnArray
      * @return array|null
-     **/
+     * */
     public function get($xpathExpression, $alwaysReturnArray = false) {
+        if ($this->useCaching && $alwaysReturnArray) {
+            if (!array_key_exists($xpathExpression, $this->cache["alwaysArray"])) {
```

Comment:
There's no need to cache the $alwaysReturnArray because it's only wrapped in an array. Better to do the wrapping on-the-fly, so you don't have to same thing cached twice.